pkgs/by-name/wa/waybar-lyric/package.nix

```
@@ -0,0 +1,31 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+}:
+buildGoModule (finalAttrs: {
+  pname = "waybar-lyric";
+  version = "0.10.0";
+
+  src = fetchFromGitHub {
+    owner = "Nadim147c";
+    repo = "waybar-lyric";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-glY0W/fB57pYrAhIxYSnkJBmqeN6vy0cwrS71CnKeK4=";
+  };
+
+  vendorHash = "sha256-DBtSC+ePl6dvHqB10FyeojnYoT3mmsWAnbs/lZLibl8=";
+
+  # Disable tests
+  # Tests fail with error "ERROR Failed to create cache directory", although program works as intended
+  doCheck = false;
+
```

I would highly recommend adding an updateScript.

This will be run periodically by bots, and can also be run by you using:

```sh
nix-shell ./maintainers/scripts/update.nix --argstr package waybar-lyric
```

or

```sh
nix-shell ./maintainers/scripts/update.nix --argstr maintainer vanadium5000
```

---

It is also a good idea to add a `--version` check:


This validates the exe can actually run, and also validates that it prints the expected version.

---


```suggestion
  doInstallCheck = true;
  nativeInstallCheckInputs = [ versionCheckHook ];
  versionCheckProgramArg = "--version";
  versionCheckKeepEnvironment = [ "XDG_CACHE_HOME" ];
  preInstallCheck = ''
    # ERROR Failed to find cache directory
    export XDG_CACHE_HOME=$(mktemp -d)
  '';
  
  passthru.updateScript = nix-update-script { };

```